worlds/ahit/DeathWishRules.py

```diff
@@ -0,0 +1,549 @@
+    "Seal the Deal":            70,
+}
+
+required_snatcher_coins = {
+    "Snatcher Coins in Mafia Town": ["Snatcher Coin - Top of HQ", "Snatcher Coin - Top of Tower",
+                                     "Snatcher Coin - Under Ruined Tower"],
+
+    "Snatcher Coins in Battle of the Birds": ["Snatcher Coin - Top of Red House", "Snatcher Coin - Train Rush",
+                                              "Snatcher Coin - Picture Perfect"],
+
+    "Snatcher Coins in Subcon Forest": ["Snatcher Coin - Swamp Tree", "Snatcher Coin - Manor Roof",
+                                        "Snatcher Coin - Giant Time Piece"],
+
+    "Snatcher Coins in Alpine Skyline": ["Snatcher Coin - Goat Village Top", "Snatcher Coin - Lava Cake",
+                                         "Snatcher Coin - Windmill"],
+
+    "Snatcher Coins in Nyakuza Metro": ["Snatcher Coin - Green Clean Tower", "Snatcher Coin - Bluefin Cat Train",
+                                        "Snatcher Coin - Pink Paw Fence"],
+}
+
+
+def set_dw_rules(world: "HatInTimeWorld"):
+    if "Snatcher's Hit List" not in world.excluded_dws \
+       or "Camera Tourist" not in world.excluded_dws:
+        set_enemy_rules(world)
+
+    dw_list: List[str] = []
+    if world.options.DWShuffle.value > 0:
+        dw_list = world.dw_shuffle
+    else:
+        for name in death_wishes.keys():
+            dw_list.append(name)
+
+    for name in dw_list:
+        if name == "Snatcher Coins in Nyakuza Metro" and not world.is_dlc2():
+            continue
+
+        dw = world.multiworld.get_region(name, world.player)
+        temp_list: List[Location] = []
+        main_objective = world.multiworld.get_location(f"{name} - Main Objective", world.player)
+        full_clear = world.multiworld.get_location(f"{name} - All Clear", world.player)
+        main_stamp = world.multiworld.get_location(f"Main Stamp - {name}", world.player)
+        bonus_stamps = world.multiworld.get_location(f"Bonus Stamps - {name}", world.player)
+        temp_list.append(main_objective)
+        temp_list.append(full_clear)
+
+        if world.options.DWShuffle.value == 0:
+            if name in dw_stamp_costs.keys():
+                for entrance in dw.entrances:
+                    add_rule(entrance, lambda state, n=name: get_total_dw_stamps(state, world) >= dw_stamp_costs[n])
+
+        if world.options.DWEnableBonus.value == 0:
+            # place nothing, but let the locations exist still, so we can use them for bonus stamp rules
+            full_clear.address = None
+            full_clear.place_locked_item(HatInTimeItem("Nothing", ItemClassification.filler, None, world.player))
+            full_clear.show_in_spoiler = False
+
+        # No need for rules if excluded - stamps will be auto-granted
+        if world.is_dw_excluded(name):
+            continue
+
+        # Specific Rules
+        modify_dw_rules(world, name)
+
+        main_rule: Callable[[CollectionState], bool]
+        for i in range(len(temp_list)):
```

Comment:
temp_list always contains exactly 2 locations, right? This part might be clearer with most of the loop body moved into a separate method, e.g. `add_dw_rules(main_objective, dw_requirements)` and `add_dw_rules(full_clear, dw_bonus_requirements)`

Reply:
Done